### ISSUE TITLE
Closing Files

### ISSUE THREAD

**Issue by nerdbaggy** [martin-sucha/zipserve#2]
First of all just wanted to say thanks for making this great library, its so helpful.

So I am using the library to zip files where the paths are passed via the URL. I have my own handler function that gets the data from the URL string, makes the template and then calls archive.ServeHTTP(w, r). But after its finished the files aren't closed, and the ServeHTTP isn't blocking so it seems there is no way to know when the request is finished. Any ideas on this would be great!

**Comment by martin-sucha** [comment]
Hi @nerdbaggy ,

`archive.ServeHTTP` uses [http.ServeContent](https://golang.org/pkg/net/http/#ServeContent) internally, so it is blocking (it returns after the request has been handled). So if you create the template in the request handler, you should be able to close the file(s) yourself after the call to `archive.ServeHTTP` in your handler returns.

This example works for me https://gist.github.com/martin-sucha/b1ba6416cfbf948b1ce499aaa44c8b7f

Please let me know whether the above information was helpful.

In case `archive.ServeHTTP` isn't blocking then it's a bug and please let me know steps how to reproduce it, ideally with some Go source code demonstrating the issue.

**Comment by nerdbaggy** [comment]
This is super helpful and exactly what I needed thanks! And you are right http.ServeContent is a block call. After doing some pcaps I realized then on Firefox when the download popup appears it is still downloading the content in the background. So in my testing I just put a fmt.Println() after the ServeHTTP and noticed that before I clicked save on the download popup the Println printed, so I assumed it wasn't a blocking call. Thanks again for the help!
